### ISSUE TITLE
STIPS images produce same background fluctuation pattern

### ISSUE THREAD

**Issue by benw1** [spacetelescope/STScI-STIPS#21]
When running STIPS processes in parallel using ProcessPoolExecutor, the resulting images will all have the same background pattern.  Likely there is a call to np.random, which should probably be changed to np.random.RandomState()

https://stackoverflow.com/questions/49847794/child-processes-generating-same-random-numbers-as-parent-process

**Comment by york-stsci** [comment]
I can't figure out whether or not this would preserve the ability to intentionally generate the same output for the same calculation in two runs by setting seed to be the same. That said, if you change the seed value for each parallel process, would that solve your immediate problem?

**Comment by benw1** [comment]
Setting the seed to be different across parallel runs does not solve this if one uses the ProcessPoolExecutor.  For now, I've gotten around this by not using that python tool; however, it seems worth considering updating the random number calls to avoid this problem if possible.  This is not a very high priority, especially if it is documented so that it does not cause too much trouble.  It can be a huge problem for people trying to simulate multiband imaging, as it leads to many false peaks in multiband stacks.

**Comment by york-stsci** [comment]
Okay. I've now moved over random calls to include RandomState.

**Comment by benw1** [comment]
I'm getting an error with this version:

Traceback (most recent call last):
  File "run_stips_bgonly.py", line 32, in <module>
    obm.nextObservation()
  File "/local/tmp/miniconda3/envs/forSTIPS3/lib/python3.6/site-packages/stips-1.0.0rc6-py3.6.egg/stips/observation_module/observation_module.py", line 187, in nextObservation
    self.instrument.reset(ra, dec, pa, filter, self.obs_count)
  File "/local/tmp/miniconda3/envs/forSTIPS3/lib/python3.6/site-packages/stips-1.0.0rc6-py3.6.egg/stips/instruments/instrument.py", line 137, in reset
    self.resetDetectors()
  File "/local/tmp/miniconda3/envs/forSTIPS3/lib/python3.6/site-packages/stips-1.0.0rc6-py3.6.egg/stips/instruments/instrument.py", line 170, in resetDetectors
    distortion=distortion, prefix=self.prefix, seed=self.seed,
AttributeError: 'WFI' object has no attribute 'seed'

**Comment by york-stsci** [comment]
The most recent push should fix this.

**Comment by benw1** [comment]
Yes.  I do not get that error with the version you just pushed. Thanks!

**Comment by york-stsci** [comment]
Does that solve this issue for you?

**Comment by benw1** [comment]
I'm getting an error:
    crPoisson = np.random.RandomSeed(seed=seed).poisson(lam=crProb, size=xSize*ySize)
AttributeError: module 'numpy.random' has no attribute 'RandomSeed'

**Comment by york-stsci** [comment]
Oops. Spelling error. Should have been RandomState. Fixed.

**Comment by benw1** [comment]
Appears to run if I change these to RandomState

**Comment by york-stsci** [comment]
The version pushed about half an hour ago to square_pixel_fix has that exact change in it.

**Comment by york-stsci** [comment]
Is this issue fixed?

**Comment by benw1** [comment]
I think so.  Has square_pixel_fix been merged into master?

**Comment by york-stsci** [comment]
Not yet. Once I'm done with this issue I'll do the merge.

**Comment by york-stsci** [comment]
Completed and merged into master.
